lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -503,6 +497,18 @@ def _on_peer_relation_departed(self, event: RelationDepartedEvent):
         else:
             event.defer()
 
+        if not self.unit.is_leader():
+            return
+
+        # Now, we register in the leader application the presence of departing unit's name
+        # We need to save them as we have a count limit
+        for attempt in Retrying(stop=stop_after_attempt(6), wait=wait_fixed(10)):
+            with attempt:
+                if not self.alt_hosts:
+                    # No neighbors found, it means this is the last unit to go away.
+                    break
+                self.opensearch_exclusions.add_exclusions(unit_name=event.departing_unit)
```

Comment:
As the `event.departing_unit` is going away, we need to register it in the app-level peer relation. Otherwise, we may not be able to clean it up.